Achievements/SelfMade.lua

```diff
@@ -19,34 +19,44 @@ end
 function self_made_achievement:Unregister()
 	self_made_achievement:UnregisterEvent("PLAYER_EQUIPMENT_CHANGED")
 end
+
+-- This function loops through a given tooltip when the "PLAYER_EQUIPMENT_CHANGED" event fires.
+-- If it finds the player's name in the tooltip it sets the varible to true and you do not fail the achievement, if false then the rest of the event fires, failing the player.
+-- This will also check if the player has a fishing pole equipped, if so they will not fail, but they must equip a selfmade weapon after that
+
 local function isSelfCreated(...)
-	for i = 1, select("#", ...) do
-		local region = select(i, ...)
-		if region and region:GetObjectType() == "FontString" then
-			local text = region:GetText() -- string or nil
-			if text ~= nil then
-				local player_name, _ = UnitName("player")
-				if string.find(text, player_name) ~= nil then
-					return true
-				end
-			end
+	local player_found = false
+	for i = 1, GameTooltip:NumLines() do
+		local player = UnitName("player")
+		local fishingPole = "Fishing Pole"
+		local mytextLeft = _G["GameTooltipTextLeft" .. i]
+		local textL = mytextLeft:GetText()
+		local mytextRight = _G["GameTooltipTextRight" .. i]
+		local textR = mytextRight:GetText()
+		if textR == fishingPole or string.match(textL, player) then
+			player_found = true
+			break
+		else
+			player_found = false
 		end
 	end
-	return false
+	return player_found
 end
 
 -- Register Definitions
+-- This executes on player equipping and item, then calls isSelfCreated() to check the tooltip for the player's name.
 self_made_achievement:SetScript("OnEvent", function(self, event, ...)
 	local arg = { ... }
+	if arg[2] == true then
```

Comment:
No action needed, but generally the pattern has been to parse the args within the conditional since there could be several types and different variations of args. I don't see us handling more events for this one so I dont think it matters